src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -363,6 +363,11 @@ internal void AddInterfaceImpl(Type iface)
 
                 foreach (MethodInfo mi in iface.GetRuntimeMethods())
                 {
+                    // Skip regular/non-virtual instance methods, static methods, and methods that cannot be overriden
+                    // (methods that cannot be overriden includes default implementation of other interface methods).
+                    if (!mi.IsVirtual || mi.IsFinal)
```

Comment:
The comment calls out statics... this condition covers them, too? Presumably because they're non-virtual?

Reply:
Yep. I called them out explicitly in the comment to document things that the `if` check below implies, but is not immediately visible in code. We don't need to filter virtual statics because there's no such thing.